packages/connect/src/api/signTransaction.ts

```diff
@@ -292,6 +294,17 @@ export default class SignTransaction extends AbstractMethod<'signTransaction', P
             await device.getCommands().unlockPath(params.unlockPath);
         }
 
+        const getHDNode = (address_n: number[]) =>
+            device
+                .getCommands()
+                .getHDNode({ address_n }, { coinInfo, unlockPath: params.unlockPath });
+
+        const outputScripts: Awaited<ReturnType<typeof deriveOutputScript>>[] = [];
+
+        for (const output of outputs) {
+            outputScripts.push(await deriveOutputScript(getHDNode, output, coinInfo.network));
+        }
```

Comment:
Maybe a bit cleaner?
```suggestion
        const outputScripts = await promiseAllSequence(
            outputs.map(output => () => deriveOutputScript(getHDNode, output, coinInfo.network)),
        );
```